Refactor wrapper handling, cleanup JB#55272

Thaodan opened the pull request.

Thaodan committed:
[lua_scripts] Restore docs argvmods

The docs for argvmods where removed in
ea0ed5c3d362bab4b5c6d823f33582ea72d2dc68 when they were refactored.
While argvmods where refactored the docs still apply so readded them.

Thaodan committed:
Revert "[scratchbox] Hadle ldconfig invoked using absolute path. Contribute to JB#30731"

This reverts commit e322c60d4d9d06348088b2409b904f12b84c6083.
Superseded by 6ff3fed4785089baad240b6d68fc0333ca04d3f1.

Thaodan committed:
[lua_scripts] Not only map argvmods to /usr/bin.

Originally argvmods where just created for ~/usr/bin~ to fill in
~lua_scripts/pathmaps/devel/00_default.lua~ 7db7147a29eb96bfb9dfee11bd0a5f80024079b3.
We need to change that so it also write rules for other dirs in
path since not all binds might be in ~/usr/bin~.

Thaodan committed:
Remove unused gcc search path

Signed-off-by: Björn Bidar <bjorn.bidar@jolla.com>

Thaodan committed:
[sb2] Remove redudant /sb2 paths from PATH. Fixes JB#55272

Wrappers are now directly mapped into the target using
argvmods.
Also indent/cleanup section setting PATH.

Signed-off-by: Björn Bidar <bjorn.bidar@jolla.com>

mkosola reviewed (changes requested):
Unnecessary change of licenses. We should keep changes as minimal as possible.

Thaodan commented:
> Unnecessary change of licenses. We should keep changes as minimal as possible.

I only changed the license of files I changes to match the project license, the rest is unchanged/adjust the license header to policy.

martyone commented on lua_scripts/argvenvp_misc.lua:
```diff
@@ -1,18 +1,46 @@
+
+   active_mapmode = m_name
+
+   for _, file in pairs(dirent.dir(session_dir.."/wrappers."..active_mapmode)) do
```
This should work:

```suggestion
   for file in dirent.files(session_dir.."/wrappers."..active_mapmode) do
```

More occurrences elsewhere(?)

Thaodan replied:
It doesn't work with pairs. `for iterator 'for iterator' is not callable (a table value))`

martyone commented on lua_scripts/argvenvp_misc.lua:
```diff
@@ -1,18 +1,46 @@
+
+for m_index,m_name in pairs(all_modes) do
+
+   active_mapmode = m_name
```
hm, is this indirection necessary or just to confuse the possible enemies? :)

Thaodan replied:
It's to follow the rest of the code.

martyone replied:
In this case it is really rather confusing. It's not like in the other cases where it exists as a sort of API for the rulefiles. I would also clearly state that `m_index` is unused employing `_` instead.

martyone commented on utils/sb2:
```diff
@@ -448,17 +448,11 @@ function sboxify_environment()
-		TMP_PATH=$TMP_PATH:$CCACHE_TOOLS_PATH
-	    fi
-	    export PATH=$TMP_PATH:$SBOX_DIR/bin:$PATH:/sbin:/usr/sbin:$SBOX_TARGET_ROOT/bin:$SBOX_TARGET_ROOT/usr/bin:$SBOX_TARGET_ROOT/usr/local/bin
+	    PATH=$HOME/.scratchbox2/$SBOX_TARGET/bin:${CCACHE_TOOLS_PATH+$CCACHE_TOOLS_PATH:}$SBOX_DIR/bin:$PATH:/sbin:/usr/sbin:$SBOX_TARGET_ROOT/bin:$SBOX_TARGET_ROOT/usr/bin:$SBOX_TARGET_ROOT/usr/local/bin
```
Wrapping would be nice

```suggestion
 	    TMP_PATH=$HOME/.scratchbox2/$SBOX_TARGET/bin
 	    TMP_PATH+=${CCACHE_TOOLS_PATH+:$CCACHE_TOOLS_PATH}
 	    TMP_PATH+=:$SBOX_DIR/bin
 	    TMP_PATH+=:$PATH
 	    TMP_PATH+=:/sbin
 	    TMP_PATH+=:/usr/sbin
 	    TMP_PATH+=:$SBOX_TARGET_ROOT/bin
 	    TMP_PATH+=:$SBOX_TARGET_ROOT/usr/bin
 	    TMP_PATH+=:$SBOX_TARGET_ROOT/usr/local/bin
 	    PATH=$TMP_PATH
```

martyone commented on lua_scripts/init_argvmods_rules.lua:
```diff
@@ -247,27 +247,20 @@ end
 
 -- first "misc":
 argvmods = {}
-argvmods_source_file = session_dir .. "/lua_scripts/argvenvp_misc.lua"
-do_file(argvmods_source_file)
-local num_rules = check_and_count_rules(argvmods)
-argvmods_to_ruletree("misc", num_rules, argvmods)
-argvmods_to_file(session_dir .. "/argvmods_misc.lua", num_rules, argvmods)
-if (debug_messages_enabled) then
-	sblib.log("debug",
-		string.format("%d rules", num_rules))
-end
+local num_rules, argvenvp_rule
 
 -- Next "gcc" and "misc";
 -- misc rules are already in argvmods table, don't clear it
-argvmods_source_file = session_dir .. "/lua_scripts/argvenvp_gcc.lua"
-do_file(argvmods_source_file)
-local num_rules = check_and_count_rules(argvmods)
-argvmods_to_ruletree("gcc", num_rules, argvmods)
-clear_attr_list()
-argvmods_to_file(session_dir .. "/argvmods_gcc.lua", num_rules, argvmods)
-if (debug_messages_enabled) then
-	sblib.log("debug",
+for _, argvenvp_rule in pairs({"misc", "gcc"}) do
+   do_file(session_dir .. "/lua_scripts/argvenvp_"..argvenvp_rule..".lua")
+   num_rules = check_and_count_rules(argvmods)
+   argvmods_to_ruletree(argvenvp_rule, num_rules, argvmods)
+   argvmods_to_file(session_dir .. "/argvmods_"..argvenvp_rule..".lua", num_rules, argvmods)
+   clear_attr_list()
+   if (debug_messages_enabled) then
+      sblib.log("debug",
 		string.format("%d rules", num_rules))
+   end
 end
```
Honestly, unless we expect another 3 rules in addition to misc and gcc, I would prefer keeping the old flat copy-paste code as a more readable option :) I hope I didn't miss any functional change or so.

Thaodan replied:
I would prefer this one since there is less code and it is easier to read.

vigejolla replied:
I have to say that I like this version more, *if* there isn't any functional change. There is some change in the order of things, but I don't know if that affects the end result.

Thaodan replied:
The order is the same and the functionality too.

vigejolla commented on lua_scripts/init_autogen_usr_bin_rules.lua:
```diff
@@ -1,7 +1,8 @@
 -- Author: Lauri T. Aarnio
 --
--- Licensed under MIT license
+-- License: LGPL-2.1
```
License change from MIT to LGPL?

Thaodan replied:
The main license of the app, I figured that when we touch the code we should align the lua code that's under MIT.

vigejolla replied:
I'm not sure that's really a good idea. While technically it's not illegal (note: I'm not a lawyer, so don't blame me when someone comes and sues us), it only serves to create confusion. I mean, the lua script was licensed to us under the MIT license. The fact that we change the text in the file to say "LGPL" doesn't change the fact that it's still under MIT license. The next person who gets to read the code just does not know it, unless they read through the git history.

Thaodan committed:
[packaging] Remove unused fakeroot dependency. Fixes JB#55486

Fakeroot dependency was replaced in:
157b6efe69f0d1bc81303548ceaf951ab2ef1279

Thaodan committed:
[packaging] Use %configure, %make in spec file

Thaodan committed:
[packaging] Fix rpm license tag. JB#50383

Thaodan committed:
[sb2] Don't use relative paths for include path and makefiles

martyone commented on utils/sb2:
```diff
@@ -448,17 +448,14 @@ function sboxify_environment()
+	    new_PATH=$HOME/.scratchbox2/$SBOX_TARGET/bin
+            new_PATH=$new_PATH:${CCACHE_TOOLS_PATH+$CCACHE_TOOLS_PATH:}$SBOX_DIR/bin:$PATH
+            PATH=$PATH:/sbin:/usr/sbin:$SBOX_TARGET_ROOT/bin
+            PATH=$PATH:$SBOX_TARGET_ROOT/usr/bin:$SBOX_TARGET_ROOT/usr/local/bin
```
o_O

martyone commented on lua_scripts/argvenvp_misc.lua:
```diff
@@ -1,18 +1,44 @@
+
+for _,m_name in pairs(all_modes) do
+
+   for _, file in pairs(dirent.dir(session_dir.."/wrappers."..m_name)) do
```
> It doesn't work with pairs. for iterator 'for iterator' is not callable (a table value))

Interesting, when I tried it forked just well:

```
for file in dirent.files(".") do print(file) end
```

Thaodan committed:
[lua_scripts] Refactor processing of argvenvp rules

Thaodan committed:
[lua_scripts] Handle wrappers in argvenvp_misc directly. JB#55272

Add argvenvp_wrappers to directly map wrappers into a target instead
of using `/sb2/wrappers` and adding it to path.

mkosola commented:
> > Unnecessary change of licenses. We should keep changes as minimal as possible.
> 
> I only changed the license of files I changes to match the project license, the rest is unchanged/adjust the license header to policy.

Yeah, but we should not touch to license headers if we have not done those. This is unnecessary change https://github.com/sailfishos/scratchbox2/pull/6/commits/c8d0017f30aefd0ef9f9ab4eebb630d5131e2cff

mkosola commented on utils/sb2:
```diff
@@ -448,17 +448,14 @@ function sboxify_environment()
+	    PATH+=$SBOX_TARGET_ROOT/usr/bin:$SBOX_TARGET_ROOT/usr/local/bin
 	else
-		export PATH=$SBOX_MODE_PATH
+            PATH=$SBOX_MODE_PATH
```
Hmm, is indentation proper here?

vigejolla reviewed (approved):
lgtm

martyone commented on utils/sb2:
```diff
@@ -448,17 +448,14 @@ function sboxify_environment()
+		new_PATH=$HOME/.scratchbox2/$SBOX_TARGET/bin
+		new_PATH+=:${CCACHE_TOOLS_PATH+$CCACHE_TOOLS_PATH:}$SBOX_DIR/bin:$PATH
+		PATH=$new_PATH:/sbin:/usr/sbin:$SBOX_TARGET_ROOT/bin
+		PATH+=$SBOX_TARGET_ROOT/usr/bin:$SBOX_TARGET_ROOT/usr/local/bin
```
Leading colon is missing here :( As a result, `$SBOX_TARGET_ROOT/bin` and `$SBOX_TARGET_ROOT/usr/bin` are essentially missing from PATH, as in

```
PATH=/home/mersdk/.scratchbox2/SailfishOS-4.4.0.58-i486.default/bin:/usr/bin:/usr/local/bin:/bin:/usr/bin:/sbin:/usr/sbin:/srv/mer/targets/SailfishOS-4.4.0.58-i486.default/bin/srv/mer/targets/SailfishOS-4.4.0.58-i486.default/usr/bin:/srv/mer/targets/SailfishOS-4.4.0.58-i486.default/usr/local/bin
```

Thaodan replied:
#12
